specs/protocol/tx_format.md

```diff
@@ -239,13 +241,40 @@ Note: when verifying a predicate, `txID`, `outputIndex`, `balanceRoot`, `stateRo
 
 Note: when executing a script, `txID`, `outputIndex`, `balanceRoot`, and `stateRoot` are initialized to the transaction ID, output index, amount, and state root of the contract with ID `contractID`, and `txPointer` is initialized to zero.
 
+### InputMessage
+
+| name                  | type        | description                                                            |
+|-----------------------|-------------|------------------------------------------------------------------------|
+| `messageID`           | `byte[32]`  | The messageID as described [here](./identifiers.md#message-id).        |
+| `sender`              | `byte[32]`  | The address of the message sender.                                     |
+| `recipient`           | `byte[32]`  | The address of the message recipient.                                  |
+| `dataLength`          | `uint16`    | Length of message data, in bytes.                                      |
+| `data`                | `byte[]`    | The message data or [abi encoded](./abi.md) call to execute.           |
```

Comment:
We don't really need Solidity ABI encoding here I think? It would be Fuel encoded data, but that's application logic. The protocol rules don't care about what the data is.
```suggestion
| `data`                | `byte[]`    | The message data .           |
```

Reply:
It's referring to Fuels abi with the thought being that most messages will be going through a standard relayer contract where the message data will be the abi for calling a Fuel contract function. But I agree that this is application specific and shouldn't be in the spec.